engine-standalone-storage/src/engine_state.rs

```diff
@@ -0,0 +1,136 @@
+use aurora_engine_sdk::io::{StorageIntermediate, IO};
+use rocksdb::DB;
+use std::cell::{Cell, RefCell};
+
+use crate::diff::{Diff, DiffValue};
+use crate::StoragePrefix;
+
+#[derive(Debug)]
+pub enum EngineStorageValue<'a> {
+    Slice(&'a [u8]),
+    Vec(Vec<u8>),
+}
+
+impl<'a> AsRef<[u8]> for EngineStorageValue<'a> {
+    fn as_ref(&self) -> &[u8] {
+        match self {
+            Self::Slice(slice) => slice,
```

Comment:
This ends up being a double reference, which the compiler is going to drop anyways automatically, but normally clippy captures these types of issues in its current version. Would be good to add a single dereference.

Reply:
Done in https://github.com/aurora-is-near/aurora-engine/pull/375/commits/6b0b3939bae0ac609c5f6dcce6a6463aad6f6088